core/src/test/scala/medeia/generic/GenericEncoderSpec.scala

```diff
@@ -38,6 +40,14 @@ class GenericEncoderSpec extends MedeiaSpec {
       ))
   }
 
+  it should "encode fields of a case class in the right order" in {
+    val original = Simple(1, "string")
+
+    val document = original.toBsonDocument
+    document.values().asScala.head should ===(BsonInt32(1))
+    document.values().asScala.last should ===(BsonString("string"))
```

Comment:
```suggestion
    document.values.asScala should contain(theSameElementsInOrderAs(List(BsonInt32(1), BsonString("string"))))
```

Reply:
> I'll have a closer look later when I have a bit of time to invest, otherwise it seems like the idea is to propagate an accumulator argument (BsonDocument) and use `append` because that as the right order?

Exactly, BsonDocument preserves the order of appends and there is no other way to change the order that I know of.

Reply:
contain theSameElementsInOrderAs does not work for the Iterable asScala produces. changed it to:
```document.values.asScala.toList should ===(List(BsonInt32(1), BsonString("string")))```